In [1]:
import pandas as pd

/Users/jisha/.pyenv/versions/3.8.6/lib/python3.8/site-packages/pandas/compat/__init__.py:120: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)


## Clean Datasets

### FMR0

In [2]:
fmr0 = pd.read_csv('csv/fmr0.csv')

In [4]:
fmr0[fmr0.duplicated(subset=['city, state'], keep=False)]

metro_code    areaname state_alpha  2017  2018  2019  2020  2021  \
0     METRO33860M33860  Montgomery          AL   587   588   581   583   640   
4     METRO13820M13820  Birmingham          AL   665   692   706   804   817   
5     METRO13820M13820      Hoover          AL   665   692   706   804   817   
6     METRO13820M13820  Birmingham          AL   665   692   706   804   817   
7     METRO13820M13820      Hoover          AL   665   692   706   804   817   
...                ...         ...         ...   ...   ...   ...   ...   ...   
3223  METRO41980MM7440    Guaynabo          PR   430   415   431   416   436   
3224  METRO38660M38660       Ponce          PR   377   371   398   377   384   
3225  METRO41980MM7440    San Juan          PR   430   415   431   416   436   
3226  METRO41980MM7440    Guaynabo          PR   430   415   431   416   436   
3227  METRO38660M49500       Yauco          PR   357   323   354   349   361   

         city, state  
0     Montgomery, AL  
4     Birmingham, AL  
5         Hoover, AL  
6     Birmingham, AL  
7         Hoover, AL  
...              ...  
3223    Guaynabo, PR  
3224       Ponce, PR  
3225    San Juan, PR  
3226    Guaynabo, PR  
3227       Yauco, PR  

[2734 rows x 9 columns]

In [5]:
fmr0 = fmr0.drop_duplicates(subset=['city, state' ], keep='last')
print(fmr0.shape)

(910, 9)


### FMR1

In [6]:
fmr1 = pd.read_csv('csv/fmr1.csv')

In [7]:
fmr1[fmr1.duplicated(subset=['city, state'], keep=False)]

metro_code    areaname state_alpha  2017  2018  2019  2020  2021  \
0     METRO33860M33860  Montgomery          AL   682   691   696   702   766   
4     METRO13820M13820  Birmingham          AL   751   768   763   861   871   
5     METRO13820M13820      Hoover          AL   751   768   763   861   871   
6     METRO13820M13820  Birmingham          AL   751   768   763   861   871   
7     METRO13820M13820      Hoover          AL   751   768   763   861   871   
...                ...         ...         ...   ...   ...   ...   ...   ...   
3223  METRO41980MM7440    Guaynabo          PR   459   439   459   451   477   
3224  METRO38660M38660       Ponce          PR   395   381   408   388   393   
3225  METRO41980MM7440    San Juan          PR   459   439   459   451   477   
3226  METRO41980MM7440    Guaynabo          PR   459   439   459   451   477   
3227  METRO38660M49500       Yauco          PR   374   337   363   359   369   

         city, state  
0     Montgomery, AL  
4     Birmingham, AL  
5         Hoover, AL  
6     Birmingham, AL  
7         Hoover, AL  
...              ...  
3223    Guaynabo, PR  
3224       Ponce, PR  
3225    San Juan, PR  
3226    Guaynabo, PR  
3227       Yauco, PR  

[2734 rows x 9 columns]

In [9]:
fmr1 = fmr1.drop_duplicates(subset=['city, state' ], keep='last')
print(fmr1.shape)

(910, 9)


### FMR2 

In [10]:
fmr2 = pd.read_csv('csv/fmr2.csv')

In [11]:
fmr2[fmr2.duplicated(subset=['city, state'], keep=False)]

metro_code    areaname state_alpha  2017  2018  2019  2020  2021  \
0     METRO33860M33860  Montgomery          AL   822   829   825   830   908   
4     METRO13820M13820  Birmingham          AL   866   882   873   986  1002   
5     METRO13820M13820      Hoover          AL   866   882   873   986  1002   
6     METRO13820M13820  Birmingham          AL   866   882   873   986  1002   
7     METRO13820M13820      Hoover          AL   866   882   873   986  1002   
...                ...         ...         ...   ...   ...   ...   ...   ...   
3223  METRO41980MM7440    Guaynabo          PR   553   530   549   531   556   
3224  METRO38660M38660       Ponce          PR   456   438   467   442   448   
3225  METRO41980MM7440    San Juan          PR   553   530   549   531   556   
3226  METRO41980MM7440    Guaynabo          PR   553   530   549   531   556   
3227  METRO38660M49500       Yauco          PR   432   389   416   409   421   

         city, state  
0     Montgomery, AL  
4     Birmingham, AL  
5         Hoover, AL  
6     Birmingham, AL  
7         Hoover, AL  
...              ...  
3223    Guaynabo, PR  
3224       Ponce, PR  
3225    San Juan, PR  
3226    Guaynabo, PR  
3227       Yauco, PR  

[2734 rows x 9 columns]

In [12]:
fmr2 = fmr2.drop_duplicates(subset=['city, state' ], keep='last')
print(fmr2.shape)

(910, 9)


### FMR3

In [13]:
fmr3 = pd.read_csv('csv/fmr3.csv')

In [14]:
fmr3[fmr3.duplicated(subset=['city, state'], keep=False)]

metro_code    areaname state_alpha  2017  2018  2019  2020  2021  \
0     METRO33860M33860  Montgomery          AL  1054  1055  1043  1047  1148   
4     METRO13820M13820  Birmingham          AL  1163  1176  1154  1291  1303   
5     METRO13820M13820      Hoover          AL  1163  1176  1154  1291  1303   
6     METRO13820M13820  Birmingham          AL  1163  1176  1154  1291  1303   
7     METRO13820M13820      Hoover          AL  1163  1176  1154  1291  1303   
...                ...         ...         ...   ...   ...   ...   ...   ...   
3223  METRO41980MM7440    Guaynabo          PR   743   706   728   706   740   
3224  METRO38660M38660       Ponce          PR   664   637   675   638   641   
3225  METRO41980MM7440    San Juan          PR   743   706   728   706   740   
3226  METRO41980MM7440    Guaynabo          PR   743   706   728   706   740   
3227  METRO38660M49500       Yauco          PR   555   501   575   567   596   

         city, state  
0     Montgomery, AL  
4     Birmingham, AL  
5         Hoover, AL  
6     Birmingham, AL  
7         Hoover, AL  
...              ...  
3223    Guaynabo, PR  
3224       Ponce, PR  
3225    San Juan, PR  
3226    Guaynabo, PR  
3227       Yauco, PR  

[2734 rows x 9 columns]

In [15]:
fmr3 = fmr3.drop_duplicates(subset=['city, state' ], keep='last')
print(fmr3.shape)

(910, 9)


### Save FMR

In [16]:
fmr0.to_csv('csv/fmr0.csv')
fmr1.to_csv('csv/fmr1.csv')
fmr2.to_csv('csv/fmr2.csv')
fmr3.to_csv('csv/fmr3.csv')

## Fix the predictions csv
- looks like there is nothing to fix here 
910*

In [18]:
fmr0_predictions = pd.read_csv('csv/fmr0_predictions.csv')
fmr1_predictions = pd.read_csv('csv/fmr1_predictions.csv')
fmr2_predictions = pd.read_csv('csv/fmr2_predictions.csv')

In [21]:
print(fmr0_predictions.shape)
fmr0_predictions.head(15)

(13650, 6)


Unnamed: 0     city, state          ds   yhat  yhat_lower  yhat_upper
0          0.0  Montgomery, AL  2017-01-01  588.0       587.0       588.0
1          1.0  Montgomery, AL  2018-01-01  589.0       588.0       589.0
2          2.0  Montgomery, AL  2019-01-01  581.0       581.0       582.0
3          3.0  Montgomery, AL  2020-01-01  583.0       583.0       584.0
4          4.0  Montgomery, AL  2021-01-01  640.0       640.0       641.0
5          5.0  Montgomery, AL  2021-12-31  584.0       532.0       635.0
6          6.0  Montgomery, AL  2022-12-31  583.0       427.0       733.0
7          7.0  Montgomery, AL  2023-12-31  582.0       293.0       877.0
8          8.0  Montgomery, AL  2024-12-31  646.0       184.0      1112.0
9          9.0  Montgomery, AL  2025-12-31  645.0       -18.0      1298.0
10        10.0  Montgomery, AL  2026-12-31  645.0      -228.0      1494.0
11        11.0  Montgomery, AL  2027-12-31  643.0      -451.0      1691.0
12        12.0  Montgomery, AL  2028-12-31  707.0      -653.0      1935.0
13        13.0  Montgomery, AL  2029-12-31  707.0      -922.0      2198.0
14        14.0  Montgomery, AL  2030-12-31  706.0     -1209.0      2481.0